aten/src/ATen/core/TensorImpl.h

```diff
@@ -1427,8 +1412,8 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
   at::Storage storage_; // TODO: Fix visibility on me
 
 protected:
-  std::vector<int64_t> sizes_;
-  std::unique_ptr<int64_t[]> strides_; // this saves two words
+  at::SmallVector<int64_t,5> sizes_;
+  at::SmallVector<int64_t,5> strides_;
```

Comment:
Probably worth noting we could save another word or two by combining the SmallVector structs, since their size is redundant, and if we need to overflow the buffer space, we could just keep the two pointers together. But that would require implementing another SmallBiVector from scratch, so we should only do this if we're desperate.